feat: universal resource internal sharing

3Hren opened the pull request:
Goals:
- [x] Provide available resources.
- [x] Optional configurable restrictions.
- [x] Push notification.
- [x] Nodes discovery (alongside with locator) for scheduling.
- [x] Debianization.

3Hren committed:
feat: universal resource internal sharing

3Hren committed:
feat: calculate CPU

3Hren committed:
feat: restrict CPU through config, logging

3Hren committed:
feat: count memory

3Hren committed:
refactor: proper types

3Hren committed:
refactor: unique error category

3Hren committed:
refactor: decompose resources

3Hren committed:
feat: notify via unicorn

3Hren committed:
feat: debianization

3Hren commented:
@cocaine-plugins TAL.

3Hren commented:
@antmat @karitra PTAL.

karitra commented on CMakeLists.txt:
```diff
@@ -72,6 +73,7 @@ SET(PLUGINS
     postgres
     unicat
     unicorn
+	uniresis
```
style? extra tab/spaces?

karitra commented on uniresis/include/cocaine/service/uniresis.hpp:
```diff
@@ -0,0 +1,32 @@
+namespace cocaine {
+namespace service {
+
+class uniresis_t : public api::service_t , dispatch<io::uniresis_tag> {
```
Is it ok for dispatch<> to be without access specifier?

karitra commented on uniresis/src/uniresis.cpp:
```diff
@@ -0,0 +1,220 @@
+        dynamic_t::object_t result;
+
+        std::vector<dynamic_t> endpoints;
+        for (auto endpoint : this->endpoints) {
```
Possible would be optimized, but may be use 'const auto& ref'?

karitra reviewed (approved):
LGTM

3Hren committed:
fix: typos and other shit

antmat commented:
We need to check if ephemeral node has gone

3Hren committed:
refactor: subscribe for our own node

This is required to be fauld tolerant to external node removing.

However we don't recreate a node if someoue has changed (not removed) its content. This is required for manual intervention.

3Hren commented:
Rereview please.

antmat reviewed (approved):
![nice](http://kleinburd.ru/news/wp-content/uploads/2017/06/foto-prikoly_042.jpg)
